pkg/network/server.go

```diff
@@ -436,6 +439,7 @@ func (s *Server) run() {
 	)
 	defer addrTimer.Stop()
 	defer peerTimer.Stop()
+	defer close(s.runFin)
```

Comment:
Defers are stacking, and channel close should be done in the end. https://go.dev/tour/flowcontrol/13